app/models/historical_emissions/record.rb

```diff
@@ -4,5 +4,37 @@ class Record < ApplicationRecord
     belongs_to :data_source, class_name: 'HistoricalEmissions::DataSource'
     belongs_to :sector, class_name: 'HistoricalEmissions::Sector'
     belongs_to :gas, class_name: 'HistoricalEmissions::Gas'
+
+    def self.find_by_params(params)
+      records = ::HistoricalEmissions::Record.
+        includes(
+          :location,
+          :data_source,
+          :sector,
+          :gas
+        )
+
+      filters(records, params)
+    end
+
+    private_class_method
```

Comment:
I don't think that this works like this. It seems that `private_class_method` takes the method name as argument.